### PR TITLE
[PATCH API-NEXT v5] Add support for AES-192 and AES-256 keys

### PR DESCRIPTION
**Pull request by lumag**
Another crypto testsuite rework + support for longer AES keys.

### CODE REVIEW

File: include/odp/api/spec/packet.h [hunk 1]
```diff
@@ -1135,6 +1135,39 @@ int odp_packet_copy_data(odp_packet_t pkt, uint32_t dst_offset,
 int odp_packet_move_data(odp_packet_t pkt, uint32_t dst_offset,
 			 uint32_t src_offset, uint32_t len);
 
+/**
+ * Set packet data to specific value
+ *
+ * Set 'len' bytes of packet data to value 'c' starting from 'offset'.  Maximum
+ * number of bytes to compare is packet data length minus the offset. Packet is
```
Comment (Bill-Fischofer-Linaro) [inline include/odp/api/spec/packet.h]: Should be "Maximum number of bytes to set.."

File: include/odp/api/spec/packet.h [hunk 2]
```diff
@@ -1135,6 +1135,39 @@ int odp_packet_copy_data(odp_packet_t pkt, uint32_t dst_offset,
...
+ * not modified.
```
Comment (Bill-Fischofer-Linaro) [inline include/odp/api/spec/packet.h]: Should be "is not modified on error."

File: include/odp/api/spec/packet.h [hunk 3]
```diff
@@ -1135,6 +1135,39 @@ int odp_packet_copy_data(odp_packet_t pkt, uint32_t dst_offset,
...
+ *
+ * @param pkt    Packet handle
+ * @param offset Byte offset into the packet
+ * @param c      Value to set
+ * @param len    Number of bytes to compare
```
Comment (Bill-Fischofer-Linaro) [inline include/odp/api/spec/packet.h]: Number of bytes to set

File: include/odp/api/spec/packet.h [hunk 4]
```diff
@@ -1135,6 +1135,39 @@ int odp_packet_copy_data(odp_packet_t pkt, uint32_t dst_offset,
...
+ * @param pkt    Packet handle
+ * @param offset Byte offset into the packet
+ * @param data   Data address
+ * @param len    Number of bytes to compare
+ *
+ * @retval 0 on success
+ * @retval <0 on failure
+ */
+int odp_packet_cmp_data(odp_packet_t pkt, uint32_t offset,
+			const void *data, uint32_t len);
```
Comment (Bill-Fischofer-Linaro) [inline include/odp/api/spec/packet.h]: Similar to odp_packet_copy_from_mem() and odp_packet_copy_from_pkt() should this be two routines: odp_packet_cmp_mem() and odp_packet_cmp_pkt() ?

Comment (lumag) [inline include/odp/api/spec/packet.h]: I do not need to `odp_packet_cmp_pkt()` ATM, so I will just rename this function to `odp_packet_cmp_mem()`.

**Comment by lumag** [comment]
PR updated to fix documentation. `odp_packet_cmp_data()` renamed to `odp_packet_cmp_mem()`

**Comment by lumag** [comment]
Dropped new packet API.

**Comment by lumag** [comment]
@psavol could you please review this PR?

**Comment by Bill-Fischofer-Linaro** [comment]
@lumag this PR no longer contains any API changes, so @psavol review is not necessary. @muvarov I believe this should be OK to merge.

**Comment by muvarov** [comment]
Merged.

**Pull request closed without merging**